src/binding/petsc4py/src/petsc4py/typing.py

```diff
@@ -89,18 +96,119 @@
 ArrayScalar = NDArray[Scalar]
 """Array of `Scalar` numbers."""
 
+AccessModeSpec = Literal['rw', 'r', 'w'] | None
+"""Access mode specification.
+
+   Possible values are:
+     - ``'rw'`` Read-Write mode.
+     - ``'r'`` Read-only mode.
+     - ``'w'`` Write-only mode.
+     - `None` as ``'rw'``.
+
+"""
+
+InsertModeSpec = InsertMode | bool | None
+"""Insertion mode specification.
+
+   Possible values are:
+     - `InsertMode.ADD_VALUES` Add new value to existing one.
+     - `InsertMode.INSERT_VALUES` Replace existing entry with new value.
+     - `None` as `InsertMode.INSERT_VALUES`.
+     - `False` as `InsertMode.INSERT_VALUES`.
+     - `True` as `InsertMode.ADD_VALUES`.
+
+"""
+
+LayoutSizeSpec = int | tuple[int, int]
+"""`int` or 2-`tuple` of `int` describing the layout sizes.
+
+   A single `int` indicates global size.
+   A `tuple` of `int` indicates ``(localsize, globalsize)``.
+
+   See Also
+   --------
+   Sys.splitOwnership
+
+"""
+
 # --- Mat ---
 
-MatSizeType = int | tuple[int, int] | tuple[tuple[int, int], tuple[int, int]]
-MatBlockSizeType = tuple[int, int] | int
-CSRIndicesType = tuple[Sequence[int], Sequence[int]]
-CSRType = tuple[Sequence[int], Sequence[int], Sequence[int]]
-NNZType = int | Sequence[int] | tuple[Sequence[int], Sequence[int]]
+MatAssemblySpec = Mat.AssemblyType | bool | None
+"""Matrix assembly specification.
+
+   Possible values are:
+     - `Mat.AssemblyType.FINAL`
+     - `Mat.AssemblyType.FLUSH`
+     - `None` as `Mat.AssemblyType.FINAL`
+     - `False` as `Mat.AssemblyType.FINAL`
+     - `True` as `Mat.AssemblyType.FLUSH`
+
+   See Also
+   --------
+   petsc.MatAssemblyType
+
+"""
+
+MatSizeSpec = int | tuple[int, int] | tuple[tuple[int, int], tuple[int, int]]
+"""`int` or (nested) `tuple` of `int` describing the matrix sizes.
+
+   If `int` then rows = columns.
+   A single `tuple` of `int` indicates ``(rows, columns)``.
+   A nested `tuple` of `int` indicates ``((localrows, rows), (localcolumns, columns))``.
```

Comment:
Isn't' this way slightly easier to read?

```suggestion
   A nested `tuple` of `int` indicates ``((local_rows, rows), (local_columns, columns))``.
```